kb/development/mbedtls-coding-standards.md

```diff
@@ -236,28 +236,116 @@ Most functions that need cleanup have a single cleanup block at the end. The lab
 
 Mbed TLS code should minimize use of external functions. Standard `libc` functions are allowed, but should be documented in the [KB article on external dependencies](what-external-dependencies-does-mbedtls-rely-on.md).
 
+## Preprocessor
+
 ### Minimize code based on preprocessor directives
 
 To minimize the code size and external dependencies, the availability of modules and module functionality is controlled by preprocessor directives located in `mbedtls/mbedtls_config.h`. Each module should have at least its own module define for enabling or disabling the module altogether. Other files using the module header should only include the header file if the module is actually available.
 
 Since often systems that use Mbed TLS do not have a file system, functions specifically using the file system should be contained in `MBEDTLS_FS_IO` directives.
 
+### Conditional compilation hygiene
+
+The span of a conditional compilation directive should generally be a sequence of C instructions or declarations.
+
+If necessary, the span can be an expression or a sequence of expressions. For example, this is acceptable:
+```c
+static char self_test_key = {
+    0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+    0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f,
+#if !defined(SUPPORT_128_BIT_KEYS)
+    0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17,
+    0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e, 0x1f,
```

Comment:
+1 for having a trailing `,` on the final element of an array initialiser - not only does it allow things like this, it makes diffs smaller (once less line to change) when code is modified. Same for enums too.

Reply:
And avoids head scratching when extending code like
```
things = {
    "one thing",
    "and another",
    "also..."
    "this is just in",
};
```
Now why is `"this is just in"` not registering as a valid input...

Reply:
Before I forget: https://github.com/Mbed-TLS/mbedtls-docs/pull/58